src/CalldataList.sol

```diff
@@ -212,44 +224,4 @@ abstract contract CalldataList {
         /// delete the calldata list for the given contract and selector
         delete calldataList[contractAddress][selector];
     }
-
-    /// @notice remove a calldata check by calldata
-    /// @param contractAddress the address of the contract that the calldata check is removed from
-    /// @param selector the function selector of the function that the calldata check is removed from
-    /// @param startIndex the start index of the calldata
-    /// @param endIndex the end index of the calldata
-    /// @param data the calldata that is removed
-    function _removeCalldataCheck(
```

Comment:
why removing the opt to delete a single calldata?

Reply:
There is a method to remove a single calldata. But it is by array index.
```
function _removeCalldataCheck(
        address contractAddress,
        bytes4 selector,
        uint256 index
    ) 
    ```